applications/plugins/SceneCreator/SceneCreator.cpp

```diff
@@ -329,12 +345,26 @@ Node::SPtr createVisualNodeVec3(Node::SPtr  parent, MechanicalObject3::SPtr  dof
     visual->setRotation(rotation[0],rotation[1],rotation[2]);
     VisualNode->addObject(visual);
 
-    BarycentricMapping3_to_Ext3::SPtr mapping = New<BarycentricMapping3_to_Ext3>();
-    mapping->setModels(dof.get(), visual.get());
-    mapping->setName("Mapping Visual");
-    mapping->setPathInputObject(refDof);
-    mapping->setPathOutputObject(refVisual);
-    VisualNode->addObject(mapping);
+    if (mappingT == MT_Barycentric) // TODO check if possible to create a baseMapping::SPtr before the switch
+    {
+        BarycentricMapping3_to_Ext3::SPtr mapping = sofa::core::objectmodel::New<BarycentricMapping3_to_Ext3>();
+        mapping->setModels(dof.get(), visual.get());
+        mapping->setName("Mapping Visual");
+        mapping->setPathInputObject(refDof);
+        mapping->setPathOutputObject(refVisual);
+        VisualNode->addObject(mapping);
+    }
+    else if (mappingT == MT_Identity)
+    {
+        IdentityMapping3_to_Ext3::SPtr mapping = sofa::core::objectmodel::New<IdentityMapping3_to_Ext3>();
+        mapping->setModels(dof.get(), visual.get());
+        mapping->setName("Mapping Visual");
+        mapping->setPathInputObject(refDof);
+        mapping->setPathOutputObject(refVisual);
+        VisualNode->addObject(mapping);
+    }
+    else
+        std::cout << "Error: mapping visual not possible";
```

Comment:
answer for the TODO: "I don't know if it is possible as it is a BaseMapping in sofa core. 
I was going to ask that in the next meeting."

ok  for the change.